Sources/MetricsTestKit/TestMetrics.swift

```diff
@@ -486,14 +492,12 @@ public final class TestTimer: TestMetric, TimerHandler, Equatable {
         }
     }
 
-    func retrieveValueInPreferredUnit(atIndex i: Int) -> Double {
-        return self.lock.withLock {
-            let value = _values[i].1
-            guard let displayUnit = self.displayUnit else {
-                return Double(value)
-            }
-            return Double(value) / Double(displayUnit.scaleFromNanoseconds)
+    public func retrieveValueInPreferredUnit(atIndex i: Int) -> Double {
```

Comment:
I wonder if just `valueInPreferredUnit` would be fine, while making this public?